Scripts/aws_stack_provision.ps1

```diff
@@ -1,103 +1,302 @@
 ## Powershell script for
 
 param (
-    [string] $lansa_version
+[parameter(Mandatory=$true)]    [string] $lansa_version,
+[parameter(Mandatory=$false)]   [string] $tag_lansa_version ## lansa version that would be used to tag cloned stack
+
 )
 
+Write-Host "Searching for VM with Lansa Version tag = $lansa_version"
 
 $EXISTING_INSTANCE_COUNT = (((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId).count
 
 if ($EXISTING_INSTANCE_COUNT -eq 1 ){
-  echo "VM with same lansa Version tag exist"
-  echo "Checking VM Status"
-  $INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
-  $INSTANCE_STATE = (((Get-EC2InstanceStatus -IncludeAllInstance $true -InstanceId $INSTANCE_ID).InstanceState).Name).Value
-
-  if ($INSTANCE_STATE -eq "running"){
-     echo "VM is already in running state"}
-  elseif ($INSTANCE_STATE -eq "stopped"){
-     echo "VM is in stopped state"
-     echo "Starting the VM"
-     $INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
-     Start-Sleep -Seconds 60
-     $NEW_STATUS = (((Start-EC2Instance -InstanceID $INSTANCE_ID).CurrentState).Name).Value
-     echo "Current status of VM is $NEW_STATUS"}
-  else{
-     echo "VM is neither Running nor stopped"
-     echo "VM status is $INSTANCE_STATE"}  }
+  
+	Write-Host "VM with Lansa Version tag = $lansa_version exist"
+	Write-Host "Checking VM Status"
+	$INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
+	$INSTANCE_STATE = (((Get-EC2InstanceStatus -IncludeAllInstance $true -InstanceId $INSTANCE_ID).InstanceState).Name).Value
+
+  	if ($INSTANCE_STATE -eq "running"){
+		Write-Host "VM is already in running state"}
+
+	elseif ($INSTANCE_STATE -eq "stopped"){
+     		Write-Host "VM is in stopped state"
+     		Write-Host "Starting the VM"
+     		$INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
+     		$NEW_STATUS = (((Start-EC2Instance -InstanceID $INSTANCE_ID).CurrentState).Name).Value
+
+     		Write-Host "Current status of VM is $NEW_STATUS"}
```

Comment:
You should wait until you can get the password. 10 minute timeout. If it times out, throw an exception

Reply:
When we can fetch the VM password even if when VM is in Stop state So do we really need to fetch the password when we are just starting the VM.
I have added the fetch password logic when we are provisioning new VM because there it waits for the VM to be in ready state  before we would be able to fetch the password.

Reply:
The ability to fetch the password indicates the VM is ready to use. The Instance state is not relevant to its readiness.

Reply:
So even if the VM is in stop state, can we say VM is ready to use?

Reply:
You cannot fetch the password if the VM is in the stopped state.

Reply:
![image](https://user-images.githubusercontent.com/98045783/198496591-172d041e-4833-46ce-a686-75a11100ed4e.png)